tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -806,6 +806,21 @@ class ExecutionTest {
 
     VERIFY_SUCCEEDED(CreateDXGIFactory1(IID_PPV_ARGS(&factory)));
     if (GetTestParamUseWARP(UseWarpByDefault())) {
+
+      // The WARP_DLL runtime parameter can be used to specify a specific DLL to
+      // load.  To force this to be used, we make sure that this DLL is loaded
+      // before attempting to create the device.
+      HMODULE ExplicitlyLoadedWarpDll = NULL;
```

Comment:
>HMODULE ExplicitlyLoadedWarpDll = NULL; [](http://example.com/codeflow?start=6&length=39)

You can use a wil::unqiue_hmdoule here instead and avoid needing the code that free's it below. Also good because it avoids leaking the handle if the verify throws as the wil stuff is cleaned up on scope exit.

#include <wil/resource.h>

wil::unique_hmodule ExplicitlyLoadedWarpDLL(LoabLibraryExW(WarpDllPath, NULL, 0);

Reply:
I don't want to add a dependency on wil here though, but good catch on it not being exception safe. I've put a small RAII helper in here to handle that.